src/swarm-engine/Swarm/Game/Robot.hs

```diff
@@ -603,16 +616,13 @@ instance FromJSONE EntityMap TRobot where
       <*> liftE (v .:? "loc")
       <*> liftE (fmap getHeading $ v .:? "dir" .!= HeadingSpec zero)
       <*> localE (const defDisplay) (v ..:? "display" ..!= defDisplay)
-      <*> liftE (mkMachine <$> (v .:? "program"))
+      <*> liftE (v .:? "program")
```

Comment:
Yes, this is perfect!  It makes complete sense to me to store a `ProcessedTerm` in a `TemplateRobot` and only turn it into a CESK machine when instantiating it to a `ConcreteRobot`.  One of those things that seems completely obvious in retrospect.